File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int profit=0;
        int minprices=INT_MAX;
        for(int i=0;i<n;i++){
        if(prices[i]<minprices){
            minprices=prices[i];
            
        }
            else{
                profit=max(profit,prices[i]-minprices);
        }
        }
        return profit;
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let n=prices.length;
    let profit=0;
    let minprices=Number.POSITIVE_INFINITY;
    for(let i=0;i<n;i++){
        if(prices[i]<minprices){
            minprices=prices[i];   
        }
        profit=Math.max(profit,prices[i]-minprices);
    }
    return profit;
};